framework/include/hdgbcs/HDGIntegratedBC.h

```diff
@@ -0,0 +1,118 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "IntegratedBCBase.h"
+#include "HDGInterface.h"
+
+/**
+ * An integrated boundary condition for hybridized finite element formulations
+ */
+class HDGIntegratedBC : public IntegratedBCBase, public HDGInterface
+{
+public:
+  static InputParameters validParams();
+
+  HDGIntegratedBC(const InputParameters & parameters);
+
+  virtual void computeResidual() override final;
+  virtual void computeJacobian() override final;
+  virtual void computeOffDiagJacobian(unsigned int) override final;
+  virtual void computeOffDiagJacobianScalar(unsigned int) override final;
+  virtual void computeResidualAndJacobian() override final;
+
+protected:
+  void createIdentityResidual(const MooseArray<std::vector<Real>> & phi,
+                              const MooseArray<Number> & sol,
+                              const std::size_t n_dofs,
+                              const unsigned int i_offset);
+
+  void createIdentityJacobian(const MooseArray<std::vector<Real>> & phi,
+                              const std::size_t n_dofs,
+                              const unsigned int ij_offset);
+
+  /// normals at quadrature points
+  const MooseArray<Point> & _normals;
+
+  /// The auxiliary system
+  SystemBase & _aux_sys;
+
+  /// transformed Jacobian weights on the face
+  const MooseArray<Real> & _JxW_face;
+
+  /// The face quadrature rule
+  const QBase * const & _qrule_face;
+
+  /// active quadrature points on the face
+  const MooseArray<Point> & _q_point_face;
+
+  /// The current neighbor
+  const Elem * _neigh = nullptr;
+
+private:
+  friend class HDGKernel;
```

Comment:
The HDGKernel is the central repository for elemental contributions. It will actually loop over the HDG BCs and its residual/Jacobian contributions to the central repository. So it accesses HDGBC data. See:
```c++
void
HDGKernel::addBCData(const HDGIntegratedBC & hibc)
{
  _PrimalMat += hibc._PrimalMat;
  _LMMat += hibc._LMMat;
  _PrimalLM += hibc._PrimalLM;
  _LMPrimal += hibc._LMPrimal;
  _PrimalVec += hibc._PrimalVec;
  _LMVec += hibc._LMVec;
}
```
The reason we do this here and not in, say, the `Assembly` class, is that only the physics knows what fields are primal vs. LM fields and hence how to condense the primal data into LM data before adding to the global residual/Jacobian data structures which contain only the LM degrees of freedom